commands/commands.go

```diff
@@ -384,15 +384,19 @@ func CmdAuth(c *cli.Context) error {
 		}
 	}
 
-	fmt.Print("Password: ")
-	pass, err := gopass.GetPasswdMasked()
-	if err != nil {
-		return cli.NewExitError(fmt.Sprintf("error reading password: %v", err), 1)
-	}
-
-	// Validate password
+	// Take password from argument, and fall back to input
+	pass := c.String("p")
 	if len(pass) == 0 {
```

Comment:
Just a suggestion to improve things idiomatically: since here `pass` is a `string` and not a `[]byte` (as it was before), we should check if it's an empty string, rather than if the length is 0.